src/Zicht/Bundle/UrlBundle/Service/UrlValidator.php

```diff
@@ -18,55 +18,50 @@
 class UrlValidator
 {
     /**
-     * @var Client
-     */
-    private $client;
-
-    private $invalidStatusCodes;
-
-    /**
-     * UrlValidator constructor.
+     * Returns true when url does not return error codes or not found.
      *
-     * @param Client $client
+     * @param string $url
+     * @return boolean
      */
-    public function __construct(Client $client)
+    public function validate($url)
     {
-        $this->client = $client;
-        $this->invalidStatusCodes = array(
-            400,
-            401,
-            403,
-            404,
-            405,
-            500,
-            501,
-            502,
-            503,
-            504,
-            511,
-        );
+        if (null !== ($headers = $this->getHeader($url))) {
+            $statusCode = $this->getStatusCode($headers);
+            // see https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html
+            return ($statusCode >= 200 && $statusCode < 300);
+        }
+        return false;
     }
 
     /**
-     * Returns true when url does not return error codes or not found.
+     * Fetch a url with a HEAD request because we just want to check status code.
      *
      * @param string $url
-     * @return boolean
+     * @return string|bool
      */
-    public function validate($url)
+    protected function getHeader($url)
     {
-        try {
-            $response = $this->client->get($url);
-        } catch (ConnectException $e) {
-            return false;
-        } catch (RequestException $e) {
-            return false;
+        if (false !== @file_get_contents($url, false, stream_context_create(['http' => ['method' => 'HEAD']]))) {
+            return $http_response_header;
+        } else {
+            return null;
         }
+    }
 
-        if (in_array($response->getStatusCode(), $this->invalidStatusCodes)) {
-            return false;
+    /**
+     * Parse the headers array and search for the status pattern
+     *
+     * @param array $headers
+     * @return int
+     */
+    protected function getStatusCode(array $headers)
+    {
+        $status = 0;
+        foreach ($headers as $header) {
+            if (preg_match('#^HTTP/(?:[^\s]+)\s(?P<code>\d+)\s#', $header, $match)) {
+                $status = (int)$match['code'];
```

Comment:
bestaat er een mogelijkheid dat er meerdere status codes in een header zitten? Als dat niet zo is kan je hier al returnen. en de laatste return is dan een null. en dan hoef je $status nooit te initialiseren.

Reply:
docs lezen?

Reply:
Ik heb het niet kunnen vinden dat er meerdere mogelijk zijn.

Reply:
tip redirect (status 30x)

Reply:
We willen hier de laatse status code in de response, dat is wat er hier gebeurd :+1: